lldb/source/Target/Target.cpp

```diff
@@ -1608,9 +1612,28 @@ void Target::SetExecutableModule(ModuleSP &executable_sp,
         if (image_module_sp) {
           added_modules.AppendIfNeeded(image_module_sp, false);
           ObjectFile *objfile = image_module_sp->GetObjectFile();
-          if (objfile)
+          if (objfile) {
+            std::lock_guard<std::mutex> guard(dependent_files_mutex);
             objfile->GetDependentModules(dependent_files);
```

Comment:
I wonder if this operation is heavy in any of the `ObjectFile` implementations. If it is we may want to lock the mutex only when an actual append to the dependent_files happens.

Reply:
good question.  for mach-o this is a simple iteration over the load commands, but we might want to to pass a temporaryfile list object to this method, and then acquire the lock and append its entries to `dependency_files`, if we didn't want to assume that.

Reply:
Yeah fair enough. Even the Mach-O one is doing some filesystem operations that really shouldn't be happening while holding the lock. I don't think we need to make every FileSpecList thread safe, and passing an optional `mutex*` was messy, so I went with the local copy as Jason suggested.

Reply:
Just in case, did you check for a regression in performance after the latest patch?

Reply:
Yeah I did. I was hoping for a slight speedup, but the difference was in the noise.